### ISSUE TITLE
Particles don't work when too far away from player

### ISSUE THREAD

**Issue by LoneWolfHT** [minetest/minetest#9127]
##### Minetest version
```
5.1
```

##### Summary
I'm working on adding snow that falls from the sky in CTF maps and have noticed that unless I turn my viewing range up really high particles won't fall and will just get stuck like this.
![image](https://user-images.githubusercontent.com/28972859/69084894-03e47800-09fa-11ea-95ae-1654ab830721.png)

I've tested with both particlespawers and with a globalstep that spawns each particle individually. The collisiondetection setting has no effect on this weird behaviour

##### Steps to reproduce
Run this mod with viewing range at 20. The snow particles spawn at y 75 around `0, 0, 0`
```lua
minetest.register_globalstep(function(dtime)
	minpos = vector.new(-100, 75, -100)
	maxpos = vector.new(100, 75, 100)

	minetest.add_particle({
        pos = vector.new(math.random(minpos.x, maxpos.x), math.random(minpos.y, maxpos.y), math.random(minpos.z, maxpos.z)),
        velocity = vector.new(0, math.random(-15, -10), 0),
        acceleration = vector.new(),
        expirationtime = 10,
        size = math.random(4, 6),
        collisiondetection = true,
        collision_removal = true,
        object_collision = true,
        vertical = false,
        texture = "default_snow.png",
        glow = 0,
    })
end)
```

**Comment by style-nine** [comment]
Does object_collision work outside the view range?

**Comment by LoneWolfHT** [comment]
Not sure how that would fix anything?
```lua
object_collision = false,
-- If true particle collides with objects that are defined as
-- `physical = true,` and `collide_with_objects = true,`.
-- Requires collisiondetection = true to have any effect.
```

**Comment by style-nine** [comment]
In my testing, particles work just fine. That was the only thing I can think of to explain what you're seeing.

**Comment by LoneWolfHT** [comment]
> In my testing, particles work just fine. That was the only thing I can think of to explain what you're seeing.

Please test with the mod/steps I provided. Nothing is wrong with the code. Since everything works fine when my viewing range is high enough to load the particlespawner pos

**Comment by style-nine** [comment]
Okay, well I performed the test that I indicated might be the problem and it turns out that it probably is the problem. Just change all those "true"s to "false"s and it works.

**Comment by LoneWolfHT** [comment]
Did you test it without changing anything in the code too? If so: Was your viewing range at 20 and was your y pos around 8-10 at the most?
I don't have access to my computer right now so I can't test this

**Comment by style-nine** [comment]
Yeah, I can confirm what you reported. It just seems to me like the real bug here is that particles are rendered at all outside your view distance.

**Comment by LoneWolfHT** [comment]
Hmmmm. This is weird. Disabling collision detection does make this work as intended now.
I need the particles to spawn and work properly outside of the player's viewing range, and I also need them to be removed upon colliding with walkable nodes

**Comment by hecktest** [comment]
Are they colliding with ignore nodes?

**Comment by LoneWolfHT** [comment]
> Are they colliding with ignore nodes?

Not sure what's going on. I can see them just sitting there, unmoving, even when collision_removal is on. They only go away when their timer reaches it's end. So they aren't necessarily colliding

**Comment by ClobberXD** [comment]
Maybe particles don't necessarily "collide" with `ignore` and just get stuck? That's very similar to players falling on `ignore` but not taking any damage.
